lib/private/Collaboration/Collaborators/Search.php

```diff
@@ -106,7 +106,7 @@ public function search($search, array $shareTypes, $lookup, $limit, $offset) {
 	}
 
 	public function registerPlugin(array $pluginInfo) {
-		$shareType = constant(Share::class . '::' . $pluginInfo['shareType']);
+		$shareType = $pluginInfo['shareType'];
```

Comment:
So your proposal is:

```suggestion
		if (str_starts_with($pluginInfo['shareType'], 'SHARE_TYPE') {
			$shareType = constant(Share::class . '::' . $pluginInfo['shareType']);
		} else {
			$shareType = $pluginInfo['shareType'];
		}
```

Reply:
How about:
```suggestion
		if (str_starts_with($pluginInfo['shareType'], 'SHARE_TYPE') {
			$shareType = constant(IShare::class . '::' . substr($pluginInfo['shareType'], 6));
		} else {
			$shareType = $pluginInfo['shareType'];
		}
```

This way we still get rid of the old constats?

Reply:
Please also note that $shareType is a int after `constant()` but not with your code.
So should use the following on the else branch
```php
		$shareType = (int) $pluginInfo['shareType'];
```